packages/angular-recipe-app/src/app/recipes/recipe-detail/recipe-detail.component.html

```diff
@@ -1,50 +1,51 @@
-<ng-template [ngIf]="selectedRecipe" [ngIfElse]="placeholderText">
-  <div class="row">
-    <div class="col-xs-12">
-      <img
-        class="img img-responsive"
-        [src]="selectedRecipe.imageUrl"
-        [alt]="selectedRecipe.name"
-      />
-    </div>
-  </div>
-  <div class="row">
-    <div class="col-xs-12">
-      <h1>{{ selectedRecipe.name }}</h1>
-    </div>
-  </div>
-  <div class="row">
-    <div class="col-xs-12">
-      <div class="btn-group" appDropdown>
-        <button type="button" class="btn btn-primary dropdown-toggle">
-          Recipe Actions <span class="caret"></span>
-        </button>
-        <ul class="dropdown-menu">
-          <li><a href="#" (click)="addIngredients()">Shopping List</a></li>
-          <li>
-            <a href="#">Edit Recipe</a>
-          </li>
-          <li>
-            <a href="#">Delete Recipe</a>
-          </li>
-        </ul>
-      </div>
-    </div>
+<div class="row">
+  <div class="col-xs-12">
+    <img
+      class="img img-responsive"
+      [src]="selectedRecipe.imageUrl"
+      [alt]="selectedRecipe.name"
+    />
   </div>
-  <div class="row">
-    <div class="col-xs-12">{{ selectedRecipe.description }}</div>
+</div>
+<div class="row">
+  <div class="col-xs-12">
+    <h1>{{ selectedRecipe.name }}</h1>
   </div>
-  <div class="row">
-    <div class="col-xs-12">
-      <ul class="list-group">
-        <li
-          class="list-group-item"
-          *ngFor="let ingredient of selectedRecipe.ingredients"
-        >
-          {{ ingredient.name }}: {{ ingredient.amount }}
+</div>
+<div class="row">
+  <div class="col-xs-12">
+    <div class="btn-group" appDropdown>
+      <button type="button" class="btn btn-primary dropdown-toggle">
+        Recipe Actions <span class="caret"></span>
+      </button>
+      <ul class="dropdown-menu">
+        <li>
+          <a style="cursor: pointer" (click)="addIngredients()"
+            >Shopping List</a
+          >
+        </li>
+        <li>
+          <a style="cursor: pointer" routerLink="edit">Edit Recipe</a>
+        </li>
+        <li>
+          <a style="cursor: pointer">Delete Recipe</a>
         </li>
       </ul>
     </div>
   </div>
-</ng-template>
-<ng-template #placeholderText>Please select a recipe</ng-template>
+</div>
+<div class="row">
+  <div class="col-xs-12">{{ selectedRecipe.description }}</div>
+</div>
+<div class="row">
+  <div class="col-xs-12">
+    <ul class="list-group">
+      <li
+        class="list-group-item"
+        *ngFor="let ingredient of selectedRecipe.ingredients"
```

Comment:
There is a new way to iterate over collections in template and it is the new control flow: https://angular.dev/guide/templates/control-flow#repeat-content-with-the-for-block

It's more readable and it has benefits in terms of performance. Right now let's use the "conservative" `*ngFor` structural directive, since it's essential knowledge, but you can create a ticket to replace them later.